transformers/modeling_beam_search.py

```diff
@@ -0,0 +1,271 @@
+# coding=utf-8
+# Copyright (c) 2019 Yang Liu
```

Comment:
Let's not forget to update this copyright (should probably rather mention Edinburgh Uni I guess)